### PR TITLE
feat: ciphertext generation algorithm

### LINKED ISSUE #2: Generation of ciphertext given a plaintext
- [x] Pick a randomly sampled difficulty
- [x] Generate a key
  - [x] Determine length of cipheralphabet based on plaintext length and difficulty
  - [x] Distribute number of homophones to letters based on frequency
  - [x] Give random homophones (numbers) for each letter and homophone given
- [x] Encipher using a random homophone from the set of each letter's homophones

### PR DESCRIPTION
**Pull request by Jatewo**
This pull request introduces a homophonic substitution cipher implementation, including the core cipher logic, homophone assignment based on English letter frequencies, and a comprehensive suite of tests. It also enhances the Python project configuration for testing.

### Core Cipher Functionality

* Added the `Cipher` class in `cipher.py`, which validates input, generates a homophonic key based on letter frequencies and difficulty, enciphers plaintext, and provides JSON and string representations.
* Implemented the `extract_homophones`, `adjust_homophones`, and `add_noise` functions in `homophones.py` to distribute cipher symbols among letters according to their frequency, with randomness and validation.

### Testing

* Added extensive tests for the cipher and homophone logic in `test_cipher.py` and `test_homophones.py`, covering input validation, key generation, ciphertext correctness, and edge cases. [[1]](diffhunk://#diff-73ddfa1579b46b001abf7ffe4f9e34da9a505f9b82c8b6d497aca5bec5aa83a3R1-R83) [[2]](diffhunk://#diff-65e0f3ea762e26871a0b92db0c303d963cb16923060902c556d3b10dc7bb5f46R1-R89)

### Project Configuration

* Updated `pyproject.toml` to include `pytest` and `pytest-cov` as dependencies, set test paths, and enabled coverage reporting for easier test execution and quality assurance.

### CODE REVIEW

**Review by TheMOHD3886 — APPROVED** [review]
lgtm 🤙

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,14 @@ version = "0.1.0"
 description = "Add your description here"
 readme = "README.md"
 requires-python = ">=3.12"
-dependencies = []
+dependencies = ["pytest", "pytest-cov"]
```
Comment (Mafusn) [inline pyproject.toml]: These should be moved to dev-dependencies together with ruff "uv add --dev pytest pytest-cov ruff"

File: test/encipherment/test_homophones.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import pytest
+from decimal import Decimal
+
+import encipherment.homophones
+
+
+@pytest.fixture
+def homphone_dict():
```
[the remark this answers is not in the document]
Comment (Mafusn) [reply test/encipherment/test_homophones.py]: I agree 💅

**Review by Mafusn — APPROVED** [review]
Looks good, but maybe move the packages to dev-dependencies
